security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -579,6 +579,13 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 			} else {
 				secret.Token = t
 			}
+		} else {
+			tok, err := ioutil.ReadFile(sc.configOptions.JWTPath)
```

Comment:
So in the past case, I remember we will check the token expired or not. and if it is expired we do `sc.callbackWithTimeout(connKey, nil /*nil indicates close the streaming connection to proxy*/)`
and rebuild the connection to and the conn will get the token from file using same logic as here

So if you add a logic here, evert time for rotating if fetcher not exists, we will try to get the  token from file.

Wondering whether it is a duplicate process

Reply:
The old logic is simply bad - there is no point in closing the whole SDS connection just to indirectly reload the file. 
And there is no point to parse a token and check for expiration - or cache it. I will remove the wrong logic in a separate 
PR.

It seems the recently added credential fetcher also follow the (right) model.

Reply:
I am ok with always loading the token from file. If this is the case, we do not need the logic to check token expiration. Also the flag "skipParsingToken" can also be removed since we never parse token.

Reply:
yes, that's the plan. I want this PR to be tested and approved - in case something goes wrong it's easy to remove, if all is ok I'll remove the old logic. Closing the SDS connection was not ideal.

Reply:
Fixed the logic - it was also inefficient, CredFetcher called even if the secret didn't need any renewal.

Reply:
And it was called even if certs were used for refresh.